prime/src/pages/BorrowActionsPage.tsx

```diff
@@ -465,21 +463,19 @@ export default function BorrowActionsPage() {
     }
     const { token0, token1 } = marginAccount;
     const earnedFeesValues = Object.values(uniswapPositionEarnedFees);
-    const priceX96 = marginAccount.sqrtPriceX96.mul(marginAccount.sqrtPriceX96).div(BIGQ96);
-    const token0FeesEarned = new Big(earnedFeesValues.reduce((p, c) => p.add(c.token0FeesEarned), new Big(0))).mul(
-      10 ** token0.decimals
-    );
-    const token1FeesEarned = new Big(earnedFeesValues.reduce((p, c) => p.add(c.token1FeesEarned), new Big(0))).mul(
-      10 ** token1.decimals
-    );
-    const token0FeesEarnedInTermsOfToken1 = priceX96.mul(token0FeesEarned).div(BIGQ96);
+    const priceX96 = marginAccount.sqrtPriceX96.square();
+    const token0FeesEarned = earnedFeesValues.reduce((p, c) => p.add(c.token0FeesEarned), GN.zero(token0.decimals));
+    const token1FeesEarned = earnedFeesValues.reduce((p, c) => p.add(c.token1FeesEarned), GN.zero(token1.decimals));
+    const token0FeesEarnedInTermsOfToken1 = priceX96.mul(token0FeesEarned).setResolution(token1.decimals);
     const totalFeesEarnedInTermsOfToken1 = token1FeesEarned.add(token0FeesEarnedInTermsOfToken1);
-    if (totalFeesEarnedInTermsOfToken1.gt(0)) {
+    if (totalFeesEarnedInTermsOfToken1.isGtZero()) {
       if (isToken0Selected) {
-        const totalFeesEarnedInTermsOfToken0 = totalFeesEarnedInTermsOfToken1.mul(BIGQ96).div(priceX96);
-        setSwapFeesInputValue(totalFeesEarnedInTermsOfToken0.div(10 ** token0.decimals).toFixed(token0.decimals));
+        const totalFeesEarnedInTermsOfToken0 = totalFeesEarnedInTermsOfToken1
+          .div(priceX96)
+          .setResolution(token0.decimals);
```

Comment:
Nice. This is totally correct, but it may be one of the places where we truncate/lack precision (similar to the reciprocal issue we talked about). If that's the case, we may be able to fix it by doing `.setResolution(reallyHighNumber)` before the division.

Don't worry about it for now. Maybe it won't be an issue; we'll see.